packages/chart/src/Contour.ts

```diff
@@ -0,0 +1,85 @@
+import { I1DChart } from "@hpcc-js/api";
+import { contourDensity as d3ContourDensity } from "d3-contour";
+import { geoPath } from "d3-geo";
+import { XYAxis } from "./XYAxis";
+
+export class Contour extends XYAxis {
+
+    constructor() {
+        super();
+        this
+            .xAxisGuideLines_default(false)
+            .yAxisGuideLines_default(false)
+            ;
+    }
+
+    xPos(host: XYAxis, d) {
+        return host.orientation() === "horizontal" ? host.dataPos(d.label) : host.valuePos(d.value);
+    }
+
+    yPos(host: XYAxis, d) {
+        return host.orientation() === "horizontal" ? host.valuePos(d.value) : host.dataPos(d.label);
+    }
+
+    layerEnter(host: XYAxis, element, duration: number = 250) {
+        super.layerEnter(host, element, duration);
+    }
+
+    layerUpdate(host: XYAxis, element, duration: number = 250) {
+        super.layerUpdate(host, element, duration);
+
+        this._palette = this._palette.switch(this.paletteID());
+
+        const data = this.flattenData(this.layerColumns(host), this.layerData(host));
+        const contourData = d3ContourDensity()
+            .x(d => this.xPos(host, d))
+            .y(d => this.yPos(host, d))
+            .size([this.width(), this.height()])
+            .bandwidth(this.contourBandwidth())
+            (data)
+            ;
+        const _vals = contourData.map(d => d.value);
+        const minValue = Math.min.apply(this, _vals);
+        const maxValue = Math.max.apply(this, _vals);
+        const lines = element.selectAll("path").data(contourData);
+        lines.enter().append("path")
+            .merge(lines)
+            .attr("d", geoPath())
+            .attr("fill", d => this.showContourFill() ? this._palette(d.value, minValue, maxValue) : "none")
+            .attr("stroke", this.contourStrokeColor())
+            .attr("stroke-linejoin", "round")
```

Comment:
Minor:  I would have put this above the "merge" as it never changes and only needs to be set on enter.